node/chainSimulator/processComponents.go

```diff
@@ -0,0 +1,485 @@
+	scheduledTxsExecutionHandler     process.ScheduledTxsExecutionHandler
+	txsSenderHandler                 process.TxsSenderHandler
+	hardforkTrigger                  factory.HardforkTrigger
+	processedMiniBlocksTracker       process.ProcessedMiniBlocksTracker
+	esdtDataStorageHandlerForAPI     vmcommon.ESDTNFTStorageHandler
+	accountsParser                   genesis.AccountsParser
+}
+
+// CreateProcessComponentsHolder will create the process components holder
+func CreateProcessComponentsHolder(args ArgsProcessComponentsHolder) (factory.ProcessComponentsHolder, error) {
+	importStartHandler, err := trigger.NewImportStartHandler(filepath.Join(args.FlagsConfig.DbDir, common.DefaultDBPath), args.FlagsConfig.Version)
+	if err != nil {
+		return nil, err
+	}
+	totalSupply, ok := big.NewInt(0).SetString(args.EconomicsConfig.GlobalSettings.GenesisTotalSupply, 10)
+	if !ok {
+		return nil, fmt.Errorf("can not parse total suply from economics.toml, %s is not a valid value",
+			args.EconomicsConfig.GlobalSettings.GenesisTotalSupply)
+	}
+
+	mintingSenderAddress := args.EconomicsConfig.GlobalSettings.GenesisMintingSenderAddress
+	argsAccountsParser := genesis.AccountsParserArgs{
+		GenesisFilePath: args.ConfigurationPathsHolder.Genesis,
+		EntireSupply:    totalSupply,
+		MinterAddress:   mintingSenderAddress,
+		PubkeyConverter: args.CoreComponents.AddressPubKeyConverter(),
+		KeyGenerator:    args.CryptoComponents.TxSignKeyGen(),
+		Hasher:          args.CoreComponents.Hasher(),
+		Marshalizer:     args.CoreComponents.InternalMarshalizer(),
+	}
+
+	accountsParser, err := parsing.NewAccountsParser(argsAccountsParser)
+	if err != nil {
+		return nil, err
+	}
+
+	smartContractParser, err := parsing.NewSmartContractsParser(
+		args.ConfigurationPathsHolder.SmartContracts,
+		args.CoreComponents.AddressPubKeyConverter(),
+		args.CryptoComponents.TxSignKeyGen(),
+	)
+	if err != nil {
+		return nil, err
+	}
+
+	historyRepoFactoryArgs := &dbLookupFactory.ArgsHistoryRepositoryFactory{
+		SelfShardID:              args.BootstrapComponents.ShardCoordinator().SelfId(),
+		Config:                   args.Config.DbLookupExtensions,
+		Hasher:                   args.CoreComponents.Hasher(),
+		Marshalizer:              args.CoreComponents.InternalMarshalizer(),
+		Store:                    args.DataComponents.StorageService(),
+		Uint64ByteSliceConverter: args.CoreComponents.Uint64ByteSliceConverter(),
+	}
+	historyRepositoryFactory, err := dbLookupFactory.NewHistoryRepositoryFactory(historyRepoFactoryArgs)
+	if err != nil {
+		return nil, err
+	}
+
+	whiteListCache, err := storageunit.NewCache(storageFactory.GetCacherFromConfig(args.Config.WhiteListPool))
+	if err != nil {
+		return nil, err
+	}
+	// TODO check if this is needed
+	whiteListRequest, err := interceptors.NewWhiteListDataVerifier(whiteListCache)
+	if err != nil {
+		return nil, err
+	}
+
+	// TODO check if this is needed
+	whiteListerVerifiedTxs, err := createWhiteListerVerifiedTxs(&args.Config)
```

Comment:
can be a mock/disabled instance, no cache needed

Reply:
removed